src/test/HostActivationTests/Constants.cs

```diff
@@ -18,11 +18,30 @@ public static class RollForwardOnNoCandidateFxSetting
             public const string EnvironmentVariable = "DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX";
         }
 
+        public static class RollForwardSetting
+        {
+            public const string RuntimeConfigPropertyName = "rollForward";
+            public const string CommandLineArgument = "--roll-forward";
+            public const string EnvironmentVariable = "DOTNET_ROLL_FORWARD";
+
+            public const string LatestPatch = "LatestPatch";
+            public const string Minor = "Minor";
+            public const string Major = "Major";
+            public const string LatestMinor = "LatestMinor";
+            public const string LatestMajor = "LatestMajor";
+            public const string Disable = "Disable";
+        }
+
         public static class FxVersion
         {
             public const string CommandLineArgument = "--fx-version";
         }
 
+        public static class RollForwardToPreRelease
+        {
+            public const string EnvironmentVariable = "DOTNET_ROLL_FORWARD_TO_PRERELEASE";
+        }
+
```

Comment:
Why are these constants in tests and not in the code?  Some concept of independent test development?

Reply:
Constants in C++ are weird (the whole .h/.cpp thing), but I can look into it. For the most part they are really only used once in the C++ product code, so the value of having it as a constant is relatively low.

In tests on the other hand the values are used in many more places, so it makes it much more valuable to have it as a constant. (And also the convenience of C# and intellisence to not make typos and so on).